Mathlib/Analysis/SpecialFunctions/Pochhammer.lean

```diff
@@ -0,0 +1,136 @@
+  · simp [heq, convexOn_const, convex_Ici]
+  · apply MonotoneOn.convexOn_of_deriv (convex_Ici (n - 1 : ℝ))
+      continuous_descPochhammer_eval.continuousOn
+      differentiable_descPochhammer_eval.differentiableOn
+    rw [interior_Ici]
+    exact monotoneOn_deriv_descPochhammer_eval n
+
+/-- The function defined to be `0` for `(-∞, n-1)` and `descPochhammer ℝ n` for `[n-1, ∞)` is
+convex on the real line. -/
+lemma convexOn_ite_descPochhammer_eval (hn : 1 ≤ n) :
+    ConvexOn ℝ Set.univ (fun (x : ℝ) ↦ if n - 1 ≤ x then (descPochhammer ℝ n).eval x else 0) := by
+  refine ⟨convex_univ, fun x _ y _ a b ha hb hab ↦ ?_⟩
+  by_cases hx : n - 1 ≤ x <;> by_cases hy : n - 1 ≤ y
+  · have hcombo : n - 1 ≤ a*x + b*y := (le_min hx hy).trans (Convex.min_le_combo x y ha hb hab)
+    simpa [hcombo, hx, hy] using (convexOn_descPochhammer_eval n).2 hx hy ha hb hab
+  · have hcombo' : a*x + b*y ≤ a*x + b*(n - 1) :=
+      add_le_add_left (mul_le_mul_of_nonneg_left (le_of_not_le hy) hb) (a*x)
+    by_cases hcombo : n - 1 ≤ a*x + b*y
+    · trans (descPochhammer ℝ n).eval (a*x + b*(n - 1))
+      · simpa [hcombo] using monotoneOn_descPochhammer_eval n hcombo (hcombo.trans hcombo') hcombo'
+      · simpa [← descPochhammer_eval_coe_nat_of_lt (Nat.sub_one_lt_of_lt hn), Nat.cast_pred hn,
+          hx, hy] using (convexOn_descPochhammer_eval n).2 hx (le_refl (n - 1 : ℝ)) ha hb hab
+    · simpa [hcombo, hx, hy] using mul_nonneg ha (descPochhammer_nonneg hx)
+  · have hcombo' : a*x + b*y ≤ a*(n - 1) + b*y :=
+      add_le_add_right (mul_le_mul_of_nonneg_left (le_of_not_le hx) ha) (b*y)
+    by_cases hcombo : n - 1 ≤ a*x + b*y
+    · trans (descPochhammer ℝ n).eval (a*(n - 1) + b*y)
+      · simpa [hcombo] using monotoneOn_descPochhammer_eval n hcombo (hcombo.trans hcombo') hcombo'
+      · simpa [← descPochhammer_eval_coe_nat_of_lt (Nat.sub_one_lt_of_lt hn), Nat.cast_pred hn,
+          hx, hy] using (convexOn_descPochhammer_eval n).2 (le_refl (n - 1 : ℝ)) hy ha hb hab
+    · simpa [hcombo, hx, hy] using mul_nonneg hb (descPochhammer_nonneg hy)
+  · have hcombo : a*x + b*y ≤ n - 1 :=
+      (Convex.combo_le_max x y ha hb hab).trans (max_le (le_of_not_le hx) (le_of_not_le hy))
+    rcases lt_or_eq_of_le hcombo with hlt | heq
+    · simp [not_le_of_lt hlt, hx, hy]
+    · simpa [heq, hx, hy, Nat.cast_pred hn] using
+        (descPochhammer_eval_coe_nat_of_lt (Nat.sub_one_lt_of_lt hn)).le
+
+private lemma ite_descPochhammer_eval_eq_descFactorial (k n : ℕ) :
+    (if (n - 1 : ℝ) ≤ (k : ℝ) then (descPochhammer ℝ n).eval (k : ℝ) else 0)
+      = k.descFactorial n := by
+  rw [descPochhammer_eval_eq_descFactorial ℝ k n, ite_eq_left_iff, not_le, eq_comm,
+    Nat.cast_eq_zero, Nat.descFactorial_eq_zero_iff_lt, ← @Nat.cast_lt ℝ]
+  exact (sub_lt_self (n : ℝ) zero_lt_one).trans'
+
+/-- Special case of **Jensen's inequality** for `Nat.descFactorial`. -/
+theorem descPochhammer_eval_le_sum_descFactorial
+    (hn : 1 ≤ n) {ι : Type*} {t : Finset ι} (p : ι → ℕ) (w : ι → ℝ)
```

Comment:
I would probably use `n ≠ 0` here, as it's easier to supply a proof of that, but it's still equivalent.